golem/managers/proposal/plugins/buffer.py

```diff
@@ -65,7 +69,21 @@ async def _on_added(self, proposal: Proposal) -> None:
 
     async def _on_expired(self, proposal: Proposal):
         logger.debug("Rejecting expired `%r` and requesting fill", proposal)
-        await proposal.reject("Proposal no longer needed due to its near expiration.")
+
+        try:
+            await proposal.reject("Proposal no longer needed due to its near expiration.")
+        except ApiException as e:
+            message = json.loads(e.body)["message"]
+            if e.status == 400 and re.match(
+                r"^Subscription \[([^]]+)\] (wasn't found|expired).$", message
+            ):
+                logger.warning(
+                    "Manual proposal expiration seems to be called too late."
+                    " Consider shortening the expire duration."
```

Comment:
the noun is either `expiry` or `expiration`... 

```suggestion
                    " Consider shortening the expiry duration."
```